icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -23,154 +26,190 @@ class UnitsTest : public IntlTest {
     struct TestCase {
-        const char16_t *source;
-        const char16_t *target;
-        const double inputValue;
-        const double expectedValue;
+        const StringPiece source;
+        const StringPiece target;
+        const UnitsMatchingState expectedState;
     } testCases[]{
-        {u"gram", u"kilogram", 1.0, 0.001},            //
-        {u"milligram", u"kilogram", 1.0, 0.000001},    //
-        {u"microgram", u"kilogram", 1.0, 0.000000001}, //
-        {u"megawatt", u"watt", 1, 1000000},            //
-        {u"megawatt", u"kilowatt", 1.0, 1000},         //
-        {u"gigabyte", u"byte", 1, 1000000000}          //
+        {"meter", "foot", CONVERTIBLE},                                         //
+        {"kilometer", "foot", CONVERTIBLE},                                     //
+        {"hectare", "square-foot", CONVERTIBLE},                                //
+        {"kilometer-per-second", "second-per-meter", RECIPROCAL},               //
+        {"square-meter", "square-foot", CONVERTIBLE},                           //
+        {"kilometer-per-second", "foot-per-second", CONVERTIBLE},               //
+        {"square-hectare", "p4-foot", CONVERTIBLE},                             //
+        {"square-kilometer-per-second", "second-per-square-meter", RECIPROCAL}, //
     };
 
     for (const auto &testCase : testCases) {
-        assertEquals("test convert", testConvert(testCase.source, testCase.target, testCase.inputValue),
-                     testCase.expectedValue);
-    }
-}
+        UErrorCode status = U_ZERO_ERROR;
 
-void UnitsTest::testMass() {
-    IcuTestErrorCode status(*this, "Units testMass");
+        MeasureUnit source = MeasureUnit::forIdentifier(testCase.source, status);
+        MeasureUnit target = MeasureUnit::forIdentifier(testCase.target, status);
 
-    // Test Cases
-    struct TestCase {
-        const char16_t *source;
-        const char16_t *target;
-        const double inputValue;
-        const double expectedValue;
-    } testCases[]{
-        {u"gram", u"kilogram", 1.0, 0.001},      //
-        {u"pound", u"kilogram", 1.0, 0.453592},  //
-        {u"pound", u"kilogram", 2.0, 0.907185},  //
-        {u"ounce", u"pound", 16.0, 1.0},         //
-        {u"ounce", u"kilogram", 16.0, 0.453592}, //
-        {u"ton", u"pound", 1.0, 2000},           //
-        {u"stone", u"pound", 1.0, 14},           //
-        {u"stone", u"kilogram", 1.0, 6.35029}    //
-    };
+        MaybeStackVector<MeasureUnit> units;
+        units.emplaceBack(source);
+        units.emplaceBack(target);
 
-    for (const auto &testCase : testCases) {
-        assertEquals("test convert", testConvert(testCase.source, testCase.target, testCase.inputValue),
-                     testCase.expectedValue);
-    }
-}
+        const auto &conversionRateInfoList = getConversionRatesInfo(units, status);
+        auto actualSatate = checkUnitsState(source, target, conversionRateInfoList, status);
```

Comment:
Typo: "actualState"

Reply:
But maybe calling this variable "convertible" makes more sense? (It's not the state of an individual unit.)

Reply:
Done !

Reply:
I changed it to `convertibility`